self_hosted/tokenizer.oomph

```diff
@@ -34,30 +37,43 @@ export class Token(Str type, Str value):
         return self.type == type and self.value == value
 
 func get_int(Str code) -> Str:
-    let len = 0
-    while len < code.length() and "0123456789".contains(code.slice(len, len+1)):
-        len = len+1
-    return code.slice(0, len)
+    let result = ""
+    while true:
+        # TODO: this is ugly
+        let did_something = false
+        for let digit = 0; digit <= 9; digit = digit + 1:
+            if code.remove_prefix(result).get().starts_with(digit.to_string()):
+                result = result + digit.to_string()
+                did_something = true
+        if not did_something:
+            return result
 
 func extract_string(Str code, Str quotes) -> Str:
-    assert(code.starts_with(quotes))
-    let end = quotes.length()
+    let result = quotes
 
     while true:
-        if code.slice(end, code.length()).starts_with(quotes):
-            return code.slice(0, end + quotes.length())
-
-        if code.slice(end, code.length()).starts_with("\\"):
-            # Skip backslash
-            end = end + 1
-        # TODO: do,while loop (and just get rid of string weirdness!!...)
-        end = end + 1
-        while end < code.length() and code.has_continuation_byte_at(end):
```

Comment:
Speaking of `length()`, make sure there's only one length method, the one that returns unicode length